### PR TITLE
Bump `Requires PHP` value

### LINKED ISSUE #12583: Bump `Requires PHP` value
## Task Description

Bump to `Requires PHP: 7.4` in `web-stories.php`

Also bump `WEBSTORIES_MINIMUM_PHP_VERSION` in `web-stories.php`

Also bump `platform.php` in `composer.json` to 7.4

Also adjust PHPUnit compat table in GitHub Actions workflow

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Update the min requirements to PHP 7.4. Tweak composer. 

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #12583

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -67,39 +69,20 @@
...
     "sort-packages": true
   },
   "extra": {
-    "downloads": {
-      "composer-normalize": {
-        "path": "vendor/bin/composer-normalize",
-        "type": "phar",
-        "url": "https://github.com/ergebnis/composer-normalize/releases/download/2.24.0/composer-normalize.phar"
-      },
-      "php-scoper": {
```
Comment (swissspidy) [inline composer.json:89]: PHP-Scoper requires PHP 8, so we can't remove this.

**Comment by swissspidy** [comment]
Marking as **blocked** just so we don't merge this before the next release, which is slated for next week, since this change is intended to be in the release after it.

File: composer.json (hunk 1 above)
Comment (spacedmonkey) [reply composer.json:89]: It required [8.1.](https://github.com/humbug/php-scoper/blob/ee08e096d379100af32275f234951e90a10874a5/composer.json#L20).Using composer to install phpscoper here means we get an older version that is supported. One from September. See https://github.com/humbug/php-scoper/pull/679

Comment (swissspidy) [reply composer.json:89]: An _older_ version, I see. The downside is we don't use the latest version and can't upgrade it anymore.

Comment (spacedmonkey) [reply composer.json:89]: But it requires PHP 8.1, something I am not running locally and we only just support. I think running the older version is a benefit here.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**